examples/tree_2d_dgsem/elixir_euler_kelvin_helmholtz_instability_ethz.jl

```diff
@@ -0,0 +1,108 @@
+using OrdinaryDiffEq
+using Trixi
+using StableRNGs
+
+
+###############################################################################
+# semidiscretization of the compressible Euler equations
+gamma = 1.4
+equations = CompressibleEulerEquations2D(gamma)
+
+"""
+    initial_condition_kelvin_helmholtz_instability_ethz(x, t, equations::CompressibleEulerEquations2D)
+
+A version of the classical Kelvin-Helmholtz instability based on
+- Ulrik S. Fjordholm, Roger Käppeli, Siddhartha Mishra, Eitan Tadmor (2014)
+  Construction of approximate entropy measure valued
+  solutions for hyperbolic systems of conservation laws
+  [arXiv: 1402.0909](https://arxiv.org/abs/1402.0909)
+"""
+function initial_condition_kelvin_helmholtz_instability_ethz(x, t, equations::CompressibleEulerEquations2D)
+  # typical resolution 128^2, 256^2
+  # domain size is [0,+1]^2
+  # interface is sharp, but randomly perturbed
```

Comment:
> `# interface is sharp, but randomly perturbed`

No it is not randomly perturbed, at least as far as I understand the code. With the current implementation, the function initializes in each invocation the RNG and then asks the same number of random numbers from it. To illustrate it, I copied the relevant part from the IC:
```julia
julia> using StableRNGs

julia> function foo()
         rng = StableRNG(100)
         m = 1
         a1 = rand(rng, m)
         a2 = rand(rng, m)
         return a1, a2
       end
foo (generic function with 1 method)

julia> foo()
([0.20804102823205062], [0.18164161613052743])

julia> foo()
([0.20804102823205062], [0.18164161613052743])

julia> foo()
([0.20804102823205062], [0.18164161613052743])
```
Thus you need to move the `rng = StableRNG(100)` outside of the IC function or otherwise you could just use constants inside it instead of adding `StableRNGs` as a new dependency.

Reply:
But just running the random stuff once to get the numbers and copying to the code is also a nice option. This should be described in a comment, of course.

Reply:
Ah, sorry for being late to the party, I missed that this was already brought up and resolved. In this case, I propose that instead of adding another dependency (which is also annoying for the users), we add something like this to the elixir:
```julia
# The random numbers used in the initial conditions have been generated as follows:
#
# using StableRNGs
#
# rng = StableRNG(100)
# m = 10
# a1 = rand(rng, m)
# a2 = rand(rng, m)
# a1 .= a1 / sum(a1)
# a2 .= a2 / sum(a2)
# b1 = (rand(rng, m) .- 0.5) .* pi
# b2 = (rand(rng, m) .- 0.5) .* pi
```
and then just use
```julia
  a1 = [0.04457096674422902, 0.03891512410182607, 0.0030191053979293433, 0.0993913172320319,
        0.1622302137588842, 0.1831383653456182, 0.11758003014101702, 0.07964318348142958,
        0.0863245324711805, 0.18518716132585408]
  a2 = [0.061688440856337096, 0.23000237877135882, 0.04453793881833177, 0.19251530387370916,
        0.11107917357941084, 0.05898041974649702, 0.09949312336096268, 0.07022276346006465,
        0.10670366489014596, 0.02477679264318211]
  b1 = [0.06582340543754152, 0.9857886297001535, 0.8450452205037154, -1.279648120993805,
        0.45454198915209526, -0.13359370986823993, 0.07062615913363897, -1.0097986278512623,
        1.0810669017430343, -0.14207309803877177]
  b2 = [-1.1376882185131414, -1.4798197129947765, 0.6139290513283818, -0.3319087388365522,
        0.14633328999192285, -0.06373231463100072, -0.6270101051216724, 0.13941252226261905,
        -1.0337526453303645, 1.0441408867083155]
```
inside the IC.

Reply:
I am open for this change. I am a bit hesitant, because at the end of the day, the whole purpose of this example setup is to then try different perturbations of the interface as new realisations and see how it affects the simulation. 

At the moment, a very simple hack to do this is to change the number inside the constructor
rng = StableRNGs(100) to rng=StableRNGs(5)
to get a new realisation of the perturbation and see how the simulation changes... 

However, as mentioned, I am open for the change - it seems indeed, that an additional dependency might be a bit much?

Reply:
If you include the code needed to generate the data, it shouldn't be too hard to do this manually, I think.

Reply:
Ok, I put the comments about the rng generation directly into the initial condition - then everything is compactly together.

Reply:
Done